lib/utils/compose_ts.ts

```diff
@@ -558,30 +557,35 @@ async function performResolution(
 			},
 			preprocessHook,
 		);
-		// Do one task at a time (Bluebird.each instead of Bluebird.all)
-		// in order to reduce peak memory usage. Resolves to buildTasks.
-		Bluebird.each(buildTasks, (buildTask) => {
-			// buildStream is falsy for "external" tasks (image pull)
-			if (!buildTask.buildStream) {
-				return buildTask;
-			}
-			// Consume each task.buildStream in order to trigger the
-			// resolution events that define fields like:
-			//     task.dockerfile, task.dockerfilePath,
-			//     task.projectType, task.resolved
-			// This mimics what is currently done in `resin-builder`.
-			return cloneTarStream(buildTask.buildStream).then(
-				(clonedStream: tar.Pack) => {
+		(async () => {
```

Comment:
The problem is the `{ error: [reject] },` above for `MultiBuild.performResolution` and I think if you return a promise inside a `new Promise` it might immediately resolve with that promise? Either way my intention was to keep the same semantics to minimize the chance of any edge case issues